.github/workflows/publish-npm-packages.yml

```diff
@@ -42,12 +42,23 @@ jobs:
         with:
           ref: ${{ github.events.inputs.amp-version }}
       - name: Get latest scripts
-        run: wget -q "${{ env.SCRIPTS_REPO }}/${{ env.SCRIPTS_DIR }}/get-extensions.js" -O ${{ env.SCRIPTS_DIR }}/get-extensions.js
+        run: |
+          wget -q "${{ env.SCRIPTS_REPO }}/${{ env.SCRIPTS_DIR }}/get-extensions.js" -O ${{ env.SCRIPTS_DIR }}/get-extensions.js
+          wget -q "${{ env.SCRIPTS_REPO }}/${{ env.SCRIPTS_DIR }}/utils.js" -O ${{ env.SCRIPTS_DIR }}/utils.js
+          wget -q "${{ env.SCRIPTS_REPO }}/${{ env.SCRIPTS_DIR }}/get-publish-order.js" -O ${{ env.SCRIPTS_DIR }}/get-publish-order.js
       - name: Get extensions to publish
         id: get-extensions
         run: |
           EXTENSIONS=$(node ${{ env.SCRIPTS_DIR }}/get-extensions.js)
           echo "::set-output name=extensions::{\"include\":${EXTENSIONS}}"
+          DEPTH_0_EXTENSIONS=$(node ${{ env.SCRIPTS_DIR }}/get-publish-order.js 0)
```

Comment:
this seems like it could be a brittle approach - what if there are depth_4 extensions because we add an extra dependency somewhere? Could we just topo-sort the extensions and then run the build/deploy shell commands for  extensions in a worker pool? In other words, parallelize using threads instead of github workflow jobs?

Reply:
(not blocking)

Reply:
Or maybe we could keep the old matrix jobs, but also use `concurrency` groups that are dynamically set to the "depth" of the current extension? 

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency